CRADLE/CorrectBiasStored/calculateOneBP.pyx

```diff
@@ -102,103 +102,113 @@ cpdef getCoefs(model, covariates):
+		shearEnd = fragEnd + 2
+
+		if shearStart < 1:
+			shearStart = 1
+			fragStart = 3
+			analysisStart = max(analysisStart, fragStart)
+		if shearEnd > chromoEnd:
+			shearEnd = chromoEnd
+			fragEnd = shearEnd - 2
+			analysisEnd = min(analysisEnd, fragEnd)
+
+		###### GET POSITIONS WHERE THE NUMBER OF FRAGMENTS > MIN_FRAGNUM_FILTER_VALUE
+		selectedIdx, highReadCountIdx, starts = selectIdx(chromo, analysisStart, analysisEnd, ctrlBWNames, experiBWNames, highRC, minFragFilterValue)
+
+		## OUTPUT FILES
+		subfinalCtrlReadCounts = [None] * len(ctrlBWNames)
+		subfinalExperiReadCounts = [None] * len(experiBWNames)
+
+		if len(selectedIdx) == 0:
+			for names in correctedCtrlReadCounts:
+				names.append((chromo, None))
+			for names in correctedExprReadCounts:
+				names.append((chromo, None))
+			continue
+
+		hdfFileName = covariates.hdfFileName(chromo)
+		with h5py.File(hdfFileName, "r") as hdfFile:
+			for rep, bwName in enumerate(ctrlBWNames):
+				with pyBigWig.open(bwName) as bwFile:
+					rcArr = np.array(bwFile.values(chromo, analysisStart, analysisEnd))
+					rcArr[np.isnan(rcArr)] = 0.0
+					rcArr = rcArr / ctrlScaler[rep]
+
+				prdvals = np.exp(
+					np.nansum(
+						(hdfFile['covari'][(analysisStart-3):(analysisEnd-3)] * covariates.selected) * COEFCTRL[rep, 1:],
+						axis=1
+					) + COEFCTRL[rep, 0]
+				)
+				prdvals[highReadCountIdx] = np.exp(
+					np.nansum(
+						(hdfFile['covari'][(analysisStart-3):(analysisEnd-3)][highReadCountIdx] * covariates.selected) * COEFCTRL_HIGHRC[rep, 1:],
+						axis=1
+					) + COEFCTRL_HIGHRC[rep, 0]
+				)
+
+				rcArr = rcArr - prdvals
+				rcArr = rcArr[selectedIdx]
+
+				idx = np.where( (rcArr < np.finfo(np.float32).min) | (rcArr > np.finfo(np.float32).max))
+				tempStarts = np.delete(starts, idx)
+				rcArr = np.delete(rcArr, idx)
+				if len(rcArr) > 0:
+					subfinalCtrlReadCounts[rep] = coalesceSections(tempStarts, rcArr, analysisEnd, binsize)
+
+			for rep, bwName in enumerate(experiBWNames):
+				with pyBigWig.open(bwName) as bwFile:
+					rcArr = np.array(bwFile.values(chromo, analysisStart, analysisEnd))
+					rcArr[np.isnan(rcArr)] = 0.0
+					rcArr = rcArr / experiScaler[rep]
+
+				prdvals = np.exp(
+					np.nansum(
+						(hdfFile['covari'][(analysisStart-3):(analysisEnd-3)] * covariates.selected) * COEFEXP[rep, 1:],
+						axis=1
+					) + COEFEXP[rep, 0]
+				)
+				prdvals[highReadCountIdx] = np.exp(
+					np.nansum(
+						(hdfFile['covari'][(analysisStart-3):(analysisEnd-3)][highReadCountIdx] * covariates.selected) * COEFEXP_HIGHRC[rep, 1:],
+						axis=1
+					) + COEFEXP_HIGHRC[rep, 0]
+				)
+
+				rcArr = rcArr - prdvals
+				rcArr = rcArr[selectedIdx]
+
+				idx = np.where( (rcArr < np.finfo(np.float32).min) | (rcArr > np.finfo(np.float32).max))
+				tempStarts = np.delete(starts, idx)
+				rcArr = np.delete(rcArr, idx)
+				if len(rcArr) > 0:
+					subfinalExperiReadCounts[rep] = coalesceSections(tempStarts, rcArr, analysisEnd, binsize)
```

Comment:
Quick question. When is analysisEnd None? In the 'def coalesceSections', I saw you check if analysisEnd is None and I wonder when analysisEnd is None.

Reply:
I think I added the `None` handling so I can eventually replace some similar functionality in `generateNormalizedObBWs` with the `coalesceSections` function.